testsuite/tests/typing-gadts/test.ml

```diff
@@ -47,7 +47,7 @@ module Exp :
 
 module List =
   struct
-    type zero
+    type zero = private Z
```

Comment:
Why private ?
I would argue for uniformly calling the constructor with the same name as the type, i.e.
```ocaml
type zero = Zero
type 'a succ = Succ of 'a
```
Note that for GADTs you don't need the parameter to be invariant, just to occur.

Reply:
I prefer to keep these types for indexing empty, which `private` does, it helps to let you know they aren't being used as ordinary types anywhere. Good point about the naming, I'll change that.